.github/actions/setup_bazel_nix/action.yml

```
@@ -25,6 +26,55 @@ runs:
           echo "BuildBuddy API key is required when cache is enabled."
           exit 1
         fi
+        if command -v nix; then
+          echo "nixPreinstalled=true" | tee -a "$GITHUB_OUTPUT"
+        else
+          echo "nixPreinstalled=false" | tee -a "$GITHUB_OUTPUT"
+        fi
+        if command -v bazel; then
+          echo "bazelPreinstalled=true" | tee -a "$GITHUB_OUTPUT"
+        else
+          echo "bazelPreinstalled=false" | tee -a "$GITHUB_OUTPUT"
+        fi
+        if [[ -f /etc/NIXOS ]]; then
+          echo "nixOS=true" | tee -a "$GITHUB_OUTPUT"
+        else
+          echo "nixOS=false" | tee -a "$GITHUB_OUTPUT"
+        fi
+        if [[ "$RUNNER_OS" == "Linux" ]]; then
+          echo "os=linux" | tee -a "$GITHUB_OUTPUT"
+        elif [[ "$RUNNER_OS" == "Windows" ]]; then
+          echo "os=windows" | tee -a "$GITHUB_OUTPUT"
+         elif [[ "$RUNNER_OS" == "macOS" ]]; then
+          echo "os=darwin" | tee -a "$GITHUB_OUTPUT"
+        else
+            echo "$RUNNER_OS not supported"
+            exit 1
+        fi
+        if [[ "$RUNNER_ARCH" == "X64" ]]; then
+          echo "arch=amd64" | tee -a "$GITHUB_OUTPUT"
+        elif [[ "$RUNNER_ARCH" == "ARM64" ]]; then
+          echo "arch=arm64" | tee -a "$GITHUB_OUTPUT"
+        else
+            echo "$RUNNER_ARCH not supported"
+            exit 1
+        fi
+        echo "::endgroup::"
+
+    - name: Install nix
+      uses: cachix/install-nix-action@6ed004b9ccb68dbc28e7c85bee15fa93dbd214ac # v22
+      if: steps.check_inputs.outputs.nixPreinstalled == 'false'
+
+    - name: Install Bazelisk
+      shell: bash
+      if: steps.check_inputs.outputs.bazelPreinstalled == 'false' && steps.check_inputs.outputs.nixOS == 'false'
+      env:
+        OS: ${{ steps.check_inputs.outputs.os }}
+        ARCH: ${{ steps.check_inputs.outputs.arch }}
+      run: |
+        echo "::group::Install Bazelisk"
+        curl -sSfLo /usr/local/bin/bazel "https://github.com/bazelbuild/bazelisk/releases/download/v1.18.0/bazelisk-${OS}-${ARCH}"
```

```suggestion
        curl -fsSLo /usr/local/bin/bazel "https://github.com/bazelbuild/bazelisk/releases/download/v1.18.0/bazelisk-${OS}-${ARCH}"
```
Just for consistency.